mission_control/navigator_missions/nav_missions/scan_the_code.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+import txros
+import tf
+import numpy as np
+import navigator_tools
+import math as math
+import matplotlib.pyplot as plt
+import rospy
+import navigator_msgs.srv as navigator_srvs
+
+
+@txros.util.cancellableInlineCallbacks
+def main(navigator):
+    navigator.change_wrench("autonomous")
+    yield navigator.vision_request("scan_the_code_activate")
+
+    # activate stereo model fitter
+    granularity = 6
+    distance = 3
+    angle = 360/granularity * 3.14/180
+    n = [0,0]
+    # this is the absolute position, give it the relative position option
+    #base = [-5,5,0]
+    base = navigator.move.latlon() + navigator.move.position
+
+    for i in range(0,granularity+1):
+        n[0] = distance * math.cos(i*angle) ; n[1] = distance * math.sin(i*angle)
```

Comment:
I think you'll need to account for the current position of the boat when starting the circle. Right now I think if you're starting on the west side of the object, the boat is going to try and go through the object to get to the east side (0 degrees) for the first point in the circle - but correct me if I'm wrong. The PoseEditor2 circle command will start the circle from the closest point to the boats current location.